src/cart/cart.repository.mongo.ts

```diff
@@ -0,0 +1,58 @@
+import { randomUUID } from "node:crypto";
+import MongoCart, { ICart, ICartItemEntity } from "../schemas/ICart";
+import { Cart, CartBase, CartItem, Product } from "../types";
+import { CartNotFoundError } from "../exceptions/CartNotFoundError";
+import { DatabaseError } from "../exceptions/DatabaseError";
+
+export class CartRepository implements CartBase {
+  async getCart(userId: string): Promise<Cart | ICart> {
+    try {
+      const cart = await MongoCart.findOne({ userId, isDeleted: false }).exec();
+      if (cart) {
+        return cart;
+      }
+      throw new CartNotFoundError();
+    } catch (err: any) {
+      if (err.name === "CartNotFound") {
+        throw err;
+      }
```

Comment:
Since you created custom exception, it would be nice to use them in error handling.
```suggestion
    } catch (err) {
      if (err instanceof CartNotFoundError) {
        throw err;
      }
```